doc/development/06_ca_manager.md

```diff
@@ -0,0 +1,978 @@
+    pub fn init_ca(&self, handle: &Handle) -> KrillResult<()> { ... }
+}
+```
+
+This results in an initialization event to be sent to the `ca_store`, which contains
+the handle (name) and a newly generated ID certificate - the `KrillSigner` owned by
+the `CaManager` is used to generate the key pair and sign this ID certificate.
+
+
+### Get a `CertAuth`
+
+We can get an existing `CertAuth` using the following function.
+
+```rust
+impl CaManager {
+    /// Gets a CA by the given handle, returns an `Err(ServerError::UnknownCA)` if it
+    /// does not exist.
+    pub async fn get_ca(&self, handle: &Handle) -> KrillResult<Arc<CertAuth>> { ... }
+}
+```
+
+**NOTE:** We do not expose the `CertAuth` type to the public API. But we have functions
+for public types. The most interesting ones are probably:
+
+```rust
+/// # Data presentation
+///
+impl CertAuth {
+    /// Returns a `CertAuthInfo` for this, which includes a data representation
+    /// of the internal structure, in particular with regards to parent, children,
+    /// resource classes and keys.
+    pub fn as_ca_info(&self) -> CertAuthInfo { ... }
+
+    /// Returns the current RoaDefinitions for this, i.e. the intended authorized
+    /// prefixes. Provided that the resources are held by this `CertAuth` one can
+    /// expect that corresponding ROA **objects** are created by the system.
+    pub fn roa_definitions(&self) -> Vec<RoaDefinition> { ... }
+    
+    /// Returns the complete set of all currently received resources, under all parents, for
+    /// this `CertAuth`
+    pub fn all_resources(&self) -> ResourceSet { ... }
+
+    /// Returns an RFC 8183 Child Request - which can be represented as XML to a
+    /// parent of this `CertAuth`
+    pub fn child_request(&self) -> rfc8183::ChildRequest { ... }
+
+    /// Returns an RFC 8183 Publisher Request - which can be represented as XML to a
+    /// repository for this `CertAuth`
+    pub fn publisher_request(&self) -> rfc8183::PublisherRequest { ... }
+}
+```
+
+
+### Update ID Certificate
+
+```rust
+/// # CA instances and identity
+///
+impl CaManager {
+    /// Updates the self-signed ID certificate for a CA. Use this with care as
+    /// RFC 8183 only talks about initial ID exchanges in the form of XML files.
+    /// It does not talk about updating identity certificates and keys. Krill supports
+    /// that a new ID key pair and certificate is generated, and has functions to update
+    /// this for a parent, a child, a repo and a publisher, but other implementations may
+    /// not support that identities are updated after initialization.
+    pub async fn ca_update_id(&self, handle: Handle, actor: &Actor) -> KrillResult<()> {
+        let cmd = CmdDet::update_id(&handle, self.signer.clone(), actor);
+        self.send_command(cmd).await?;
+        Ok(())
+    }
+}
+```
+
+
+CA Repository Related Functions
+-------------------------------
+
+
+### CA Repository Configuration
+
```

Comment:
The following paragraph talks about RFC 8181, e.g. Krill has an "embedded" vs an "Rfc8181" remote repository, and "Even if a Repository Server on the same Krill Instance is used, it can do the full RFC 8181 protocol". If I understand correctly, the difference between embedded and non-embedded affects both RFC 8181 and RFC 8183 communication and thus this paragraph could be expanded to mention both, rather than being RFC 8181 specific.